backend/internal/observability/service.go

```diff
@@ -0,0 +1,270 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package observability provides observability capabilities for Thunder including
+// event logging, metrics collection, distributed tracing, and health monitoring.
+package observability
+
+import (
+	"errors"
+	"path/filepath"
+	"sync"
+
+	"github.com/asgardeo/thunder/internal/observability/adapter"
+	"github.com/asgardeo/thunder/internal/observability/adapter/console"
+	"github.com/asgardeo/thunder/internal/observability/adapter/file"
+	"github.com/asgardeo/thunder/internal/observability/adapter/opensearch"
+	"github.com/asgardeo/thunder/internal/observability/event"
+	"github.com/asgardeo/thunder/internal/observability/formatter"
+	jsonformatter "github.com/asgardeo/thunder/internal/observability/formatter/json"
+	"github.com/asgardeo/thunder/internal/observability/publisher"
+	"github.com/asgardeo/thunder/internal/observability/subscriber/defaultsubscriber"
+	"github.com/asgardeo/thunder/internal/system/config"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+const loggerComponentName = "ObservabilityService"
+
+// Service provides observability event publishing functionality.
+// This is the main entry point for the observability system.
+// It manages the lifecycle of the publisher, which in turn manages subscribers.
+//
+// Architecture:
+//   - Service (High-level): Manages lifecycle, configuration, initialization
+//   - CategoryPublisher (Low-level): Implements event publishing logic
+//   - Subscribers (Low-level): Consume events
+//
+// The service layer owns the singleton pattern and initialization logic,
+// while the publisher layer provides the implementation.
+type Service struct {
+	publisher publisher.CategoryPublisher
+	logger    *log.Logger
+	config    *Config
+	enabled   bool
+}
+
+var (
+	serviceInstance *Service
+	serviceOnce     sync.Once
+)
+
+// GetService returns the singleton instance of the observability service.
+// Uses default configuration. For custom configuration, use InitializeWithConfig first.
+func GetService() *Service {
+	serviceOnce.Do(func() {
+		serviceInstance = NewService(nil)
+	})
+	return serviceInstance
+}
+
+// InitializeWithConfig initializes the observability service with custom configuration.
+// Must be called before GetService to take effect.
+// Returns error only if FailureMode is "strict", otherwise logs error and disables observability.
+func InitializeWithConfig(cfg *Config) (*Service, error) {
+	var svc *Service
+
+	serviceOnce.Do(func() {
+		svc = newServiceWithConfig(cfg)
+		serviceInstance = svc
+	})
+
+	if serviceInstance == nil {
+		return nil, errors.New("failed to initialize observability service")
+	}
+
+	return serviceInstance, nil
+}
+
+// NewService creates and initializes a new observability service with the given configuration.
+func NewService(cfg *Config) *Service {
```

Comment:
```suggestion
func newService(cfg *Config) *Service {
```
Let's have only one exported entrypoint